### PR TITLE
Remove malloc and free

### PR DESCRIPTION
**Pull request by tryuan99**
* Embedded code should not use dynamic memory allocation (it's dangerous!), so this PR removes `malloc` and `free` from `radio.c`.

### CODE REVIEW

File: scm_v3c/radio.c [hunk 1]
```diff
@@ -158,7 +158,8 @@ void receive_packet(bool timeout) {
 // If timeout is set false, then the function may block indefinitely
 void receive_packet_length(uint8_t pkt_len, bool timeout) {
     radio_vars.radio_mode = RX_MODE;
-    radio_vars.rxPacket_len = pkt_len;
+    radio_vars.rxPacket_len =
+        pkt_len <= RX_PKT_ANY_LEN ? pkt_len : RX_PKT_ANY_LEN;
```
Comment (daniellovell) [inline scm_v3c/radio.c:162]: Do we have min/max macros? Probably more readable to use min(pkt_len, RX_PKT_ANY_LEN)

File: scm_v3c/radio.c [hunk 2]
```diff
@@ -690,15 +687,15 @@ uint32_t build_RX_channel_table(uint32_t channel_11_LC_code) {
...
-    memset(&count_targets[0], 0, sizeof(count_targets));
+    memset(count_LC, 0, sizeof(count_LC));
+    memset(count_targets, 0, sizeof(count_targets));
 
     radio_vars.rx_channel_codes[0] = channel_11_LC_code;
 
     i = 0;
     while (i < 16) {
         LC_monotonic(radio_vars.rx_channel_codes[i]);
-        // analog_scan_chain_write_3B_fromFPGA(&ASC[0]);
```
Comment (daniellovell) [inline scm_v3c/radio.c:701]: Why are these changed? I know they're commented out, but ASC doesn't seem to be touched by the scope of this malloc removal PR.

**Review by daniellovell — COMMENTED** [review]
Looks good minus few clarifications. Has it been tested?

File: scm_v3c/radio.c (hunk 1 above)
Comment (tryuan99) [reply scm_v3c/radio.c:162]: I don't think so, I'll add it in a separate PR

File: scm_v3c/radio.c (hunk 2 above)
Comment (tryuan99) [reply scm_v3c/radio.c:701]: Oh this was just a pure stylistic change

**Comment by tryuan99** [comment]
I tested this code with the `channel_cal` project, which does radio TX and RX, and it works!
